src/densities.jl

```diff
@@ -33,17 +33,27 @@ grid `basis`, where the individual k-points are occupied according to `occupatio
 
     @sync for (ichunk, chunk) in enumerate(Iterators.partition(ik_n, chunk_length))
         Threads.@spawn for (ik, n) in chunk  # spawn a task per chunk
-            ψnk_real = ψnk_real_chunklocal[ichunk]
-            ρ_loc = ρ_chunklocal[ichunk]
-
             kpt = basis.kpoints[ik]
-            G_to_r!(ψnk_real, basis, kpt, ψ[ik][:, n])
-            ρ_loc[:, :, :, kpt.spin] .+= occupation[ik][n] .* basis.kweights[ik] .* abs2.(ψnk_real)
+            #TODO: is this the right way to got? Probably rewrite compute_density for GPUArrays
```

Comment:
I think you can just leave the code as it is and just have the chunklocal array be of the correct type? threads+GPU is a fun interaction but let's assume people don't mix those for now

Reply:
Yes, that was one of the part of the code which was functional but I must admit that I didn't understand everything that was happening. It was a good first shot though.
If I understand correctly, you suggest I make an other function which does not rely on threads, so that it can be immediately GPU compatible?

Reply:
The function allocates per-thread buffers, then each thread does its stuff, and the result is accumulated. You can assume that the code is single-threaded, and put the buffer on the GPU. Just add an assert that GPU code is run single-threaded to be safe.

Reply:
Ok, that makes sense. Done!